### ISSUE TITLE
Support for Dropdown menu alignment

### ISSUE THREAD

**Issue by ayoubkhan558** [ng-bootstrap/ng-bootstrap#3269]
support for Dropdown menu alignment as in Bootstrap 4 dropdowns
![GithubBT4Dropdown](https://user-images.githubusercontent.com/20210961/60427736-be710b00-9c10-11e9-9510-040cfc3aa93d.JPG)

The code for Dropdown menu alignment will look like this 
![carbon](https://user-images.githubusercontent.com/20210961/60428231-d72df080-9c11-11e9-9d5e-35865e4383d7.png)

**Comment by ayoubkhan558** [comment]
The code for Dropdown menu alignment will look like this 
<div ngbDropdown>
  <button class="btn btn-outline-primary" id="dropdownConfig" ngbDropdownToggle>Toggle</button>
  <div ngbDropdownMenu ngbDropdownMenuRight aria-labelledby="dropdownConfig">
    <button ngbDropdownItem>Action - 1</button>
    <button ngbDropdownItem>Another Action</button>
    <button ngbDropdownItem>Something else is here</button>
  </div>
</div>

**Comment by fbasso** [comment]
Outside a navbar, placement="bottom-right" on the button should resolve your use case. See the documentation here : https://ng-bootstrap.github.io/#/positioning for more information.

Inside a navbar, the menu is static (as described in the bootstrap documentation you provided), so in this case, class="dropdown-menu dropdown-menu-right" should have worked, but unfortunately 'x-placement' is wrongly set on the menu (this is the same on Bootstrap: x-placement is not here when located in a navbar), preventing the menu to have the right display.

**Comment by benouat** [comment]
@ayoubkhan558 I am not sure what your issue refers to in the first place ? Could you be a bit more explicit please?

---

From your screenshot, I can see you are using pseudo-code such as `ngbDropdownMenuRight`? Is that your initial issue ? You would like to have such a directive to be more aligned with Bootstrap ?

If your answer to previous question is **yes** ... 
... _Unfortunately_ we are not explicitly using the same surface API as Bootstrap on that one. For `placement` of dropdown menus, we are using specific system (which BTW also exists in Bootstrap, but is not publicly available) which is documented [here](https://ng-bootstrap.github.io/#/positioning) (@fbasso already gave you that link in his reply).

So the code to actually write the dropdown you would like to have would be

```html
<div ngbDropdown placement="bottom-right">
    <button class="btn btn-primary" id="dropdown" ngbDropdownToggle>Toggle</button>
    <div ngbDropdownMenu aria-labelledby="dropdown">
        <button ngbDropdownItem>Action - 1</button>
        <button ngbDropdownItem>Another Action</button>
        <button ngbDropdownItem>Something else is here</button>
    </div>
</div>
```

**Note** the presence of `placement="bottom-right"` on the dropdown element.

---

This example, obviously is only working for what is known as a `dynamic` dropdown, i.e. a dropdown not inside a `navbar`.

Dropdowns inside `navbars` should be (and are) statically positionned, so in that _specific_ case, `placement` should not be used (or if used it won't have any effect, thanks to @fbasso PR #3270) and _original_ `dropdown-menu-right` should be used on the toggle element.

**Comment by hoeni** [comment]
In my case, (thanks to @ctamisier's cross reference here), I was finally able to fix my right aligned menu dropdown by simply adding `display="dynamic"` to the `ngbDropdown` element.

**Comment by benouat** [comment]
We also have a documentation fix in the coming via #3268. It will be much clearer once merged.
